__tests__/app.test.js

```diff
@@ -117,7 +117,49 @@ describe('GET /api/articles/:article_id', () => {
         .get('/api/articles/99999999')
         .expect(404)
         .then(({ body }) => {
-            expect(body.msg).toBe('Not found')
+            expect(body.msg).toBe('Article_id not found!')
+        })
+    })
+})
+
+describe('GET /api/articles/:article_id/comments', () => {
+    test('GET 200 sends an array of comment objects for the passed article_id with correct properties', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.comments).toHaveLength(11);
+            body.comments.forEach((comment) => {
+                expect(comment.article_id).toBe(1);
+                expect(typeof comment.comment_id).toBe('number');
+                expect(typeof comment.votes).toBe('number');
+                expect(typeof comment.created_at).toBe('string');
+                expect(typeof comment.author).toBe('string');
+                expect(typeof comment.body).toBe('string');
+            })
+        })
+    })
+    test('comment objects should be ordered with most recent first', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .then(({ body }) => {
+            expect(body.comments).toBeSortedBy('created_at', { descending: true })
+        })
+    })
+    test('GET 400 sends an error message when passed an invalid article_id', () => {
+        return request(app)
+        .get('/api/articles/notvalidid/comments')
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe('invalid id')
+        })
+    })
+    test('GET 404 sends an error message when passed a valid id but does not exist', () => {
+        return request(app)
+        .get('/api/articles/99999999/comments')
+        .expect(404)
+        .then(({ body }) => {
+            expect(body.msg).toBe('Article_id not found!')
```

Comment:
Great work handling the 2 errors. However, you are missing a happy path test here. What if the article does exist but has no comments associated with it? For example article 13? This should respond with a 200 status and and empty array. Please do add this test 🙂